src/test/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetSubAccountServiceImplTest.java

```diff
@@ -1,241 +1,224 @@
+        plantDetail = createSourceDetail(TEST_PLANT_ACCOUNT, TEST_SUB_ACCOUNT);
+    }
+
+
+
+    @Test
+    public void testWildcardPatternMatchForBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_WILDCARD, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_REGULAR_EXACT_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_PLANT_EXACT_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForBothAccountsInWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_WILDCARD_BOTH_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfBothDetails(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testExactPatternMatchForNeitherAccountInBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_PATTERN_NO_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testBothDetailsPreservedIfNoPatternsDefinedInBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(KFSConstants.EMPTY_STRING, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testBothDetailsPreservedIfParameterDoesNotExist() throws Exception {
+        setupParameterServiceToReturnNullPlantAccountParameter();
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForMultiValueBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN, EvaluationOperator.DISALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForMultiValueWhitelist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN_REGULAR_MATCH, EvaluationOperator.ALLOW);
+        assertPreservationOfRegularDetailOnly(FOR_WHITELIST);
+    }
+
+    @Test
+    public void testWildcardPatternMatchForNeitherAccountInMultiValueBlacklist() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_MULTI_PATTERN_NO_MATCH, EvaluationOperator.DISALLOW);
+        assertPreservationOfBothDetails(FOR_BLACKLIST);
+    }
+
+    @Test
+    public void testHandlingOfInvalidPatternSetup() throws Exception {
+        setupParameterServiceWithPlantAccountValue(TEST_BAD_PATTERN, EvaluationOperator.DISALLOW);
+
+        try {
```

Comment:
Given that, it seems more investigation into ExpectedException and possible alternatives is warranted, and I'm ok with this proceeding as is. Thanks Chad.